homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,268 @@
+        """Return current tilt position for all covers."""
+        return self._tilt_position
+
+    async def async_open_cover(self, **kwargs):
+        """Move the covers up."""
+        self.hass.async_add_job(open_cover, self.hass,
+                                self._covers[KEY_OPEN_CLOSE])
+
+    async def async_close_cover(self, **kwargs):
+        """Move the covers down."""
+        self.hass.async_add_job(close_cover, self.hass,
+                                self._covers[KEY_OPEN_CLOSE])
+
+    async def async_stop_cover(self, **kwargs):
+        """Fire the stop action."""
+        self.hass.async_add_job(stop_cover, self.hass,
+                                self._covers[KEY_STOP])
+
+    async def async_set_cover_position(self, **kwargs):
+        """Set covers position."""
+        position = kwargs[ATTR_POSITION]
+        self.hass.async_add_job(set_cover_position, self.hass,
+                                position, self._covers[KEY_POSITION])
+
+    async def async_open_cover_tilt(self, **kwargs):
+        """Tilt covers open."""
+        self.hass.async_add_job(open_cover_tilt, self.hass,
+                                self._tilts[KEY_OPEN_CLOSE])
+
+    async def async_close_cover_tilt(self, **kwargs):
+        """Tilt covers closed."""
+        self.hass.async_add_job(close_cover_tilt, self.hass,
+                                self._tilts[KEY_OPEN_CLOSE])
+
+    async def async_stop_cover_tilt(self, **kwargs):
+        """Stop cover tilt."""
+        self.hass.async_add_job(stop_cover_tilt, self.hass,
+                                self._tilts[KEY_STOP])
+
+    async def async_set_cover_tilt_position(self, **kwargs):
+        """Set tilt position."""
+        tilt_position = kwargs[ATTR_TILT_POSITION]
+        self.hass.async_add_job(set_cover_tilt_position, self.hass,
+                                tilt_position, self._tilts[KEY_POSITION])
+
+    def update_attr_is_closed(self):
+        """Return if all covers are closed."""
+        if not self._covers[KEY_OPEN_CLOSE]:
+            return False
+        for entity_id in self._covers[KEY_OPEN_CLOSE]:
+            state = self.hass.states.get(entity_id)
+            if state is None or state.state != STATE_CLOSED:
+                return False
+        return True
+
+    def update_attr_cover_position(self):
+        """Return current position for all covers.
+
+        None is unknown. pos if all have the same position.
+        Else 0 if closed or 100 is fully open.
+        """
+        position = -1
+        for entity_id in self._covers[KEY_POSITION]:
+            state = self.hass.states.get(entity_id)
+            pos = state.attributes.get(ATTR_CURRENT_POSITION)
+            if position == -1:
+                position = pos
+            elif position != pos:
+                position = -1
+                break
+
+        if position != -1:
+            return position
+
+        return 0 if self.is_closed else 100
```

Comment:
The return value of this method depends on the order of execution of the calls in `async_update`; it's not real a problem, just pointing it out.

Reply:
I know. Added a comment to `async_update` to highlight this. An alternative would be to combine all `update_` functions into `async_update`, but IMO separate functions improve readability.

Reply:
Oh that is in fact true; sure, keeping units of code separated is always good. Ignore that comment :)